dropwizard-client/src/main/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactory.java

```diff
@@ -121,11 +125,28 @@ private void loadTrustMaterial(SSLContextBuilder sslContextBuilder) throws Excep
         sslContextBuilder.loadTrustMaterial(trustStore, trustStrategy);
     }
 
-    private static KeyStore loadKeyStore(String type, File path, String password) throws Exception {
-        final KeyStore keyStore = KeyStore.getInstance(type);
+    private static KeyStore loadKeyStore(String type, File path, String password,
+                                         @Nullable String provider) throws Exception {
+        KeyStore keyStore;
+
+        if (provider == null) {
+            keyStore = KeyStore.getInstance(type);
+
+        } else {
+            try {
+                keyStore = KeyStore.getInstance(type, provider);
+
+            } catch (KeyStoreException ignore) {
```

Comment:
If the configured provider doesn't work, should this exception be propagated instead of falling back -- in order to alert the user their provider is incorrect?

Reply:
I was emulating an initial implementation proposed by @arteam ....I suspect the motivation was to provide as soft an integration as possible, falling back to default providers.  
If you feel the WARN log is insufficient, however, I'm happy to propagate and enforce a more strict configuration correctness.

Reply:
I agree, that WARN should be good enough. If a keystore is still supported by the default JVM providers, I don't see a reason not to load it.